libcodechecker/server/api/authentication.py

```diff
@@ -297,3 +299,98 @@ def hasPermission(self, permission, extra_params):
                                               msg)
         finally:
             session.close()
+
+    # ============= Authorization, permission management =============
+
+    @timeit
+    def newToken(self, description):
+        """
+        Generate a new personal access token with the given description.
+        """
+        try:
+            session = self.__config_db()
+
+            token = generate_session_token()
+            user = self.getLoggedInUser()
```

Comment:
If the server doesn't have authentication enabled, this will be `""` here, and the token will be stored without a username. This can cause weird behaviour. These commands should be disabled if the server runs for guests.